server/src/main/java/org/apache/lucene/search/uhighlight/CustomUnifiedHighlighter.java

```diff
@@ -141,7 +143,21 @@ public Snippet[] highlightField(LeafReader reader, int docId, CheckedSupplier<St
             return null;
         }
         int fieldValueLength = fieldValue.length();
-        if ((offsetSource == OffsetSource.ANALYSIS) && (fieldValueLength > maxAnalyzedOffset)) {
+
+        if (fieldMaxAnalyzedOffset > maxAnalyzedOffset) {
+            throw new IllegalArgumentException(
+                "max_analyzer_offset has exceeded [" + maxAnalyzedOffset + "] - maximum allowed to be analyzed for highlighting. "
+            );
+        }
+        // throws an execption if the value is invalid, negative values are not valued with the exception of -1 wich is used to mark that it
+        // is disabled
+        if (fieldMaxAnalyzedOffset < -1) {
```

Comment:
Just looking at this particular check, I think we should fail earlier and not accept query with invalid `fieldMaxAnalyzedOffset` values ( < 0). Even -1 should not be allowed I think (it is implicit default when not passed).

Reply:
Where do you suggest that we throw the exception?

Reply:
https://github.com/opensearch-project/OpenSearch/pull/3893/files#diff-bd42ab0e216e3809be17bb788bb0886ac633a6d968de8889e288dce15e033a2aR556, at the HighlightBuilder / AbstractHighlighterBuilder

Reply:
I think that now it is better, I changed the `fieldMaxAnalyzedOffset `to integer, so it can fallback to the previous behavior when it is null and throws errors when it is negative or greater than `maxAnalyzedOffset`.